src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -331,30 +332,32 @@ private static byte[] UrlEncode(byte[] bytes, int offset, int count)
 
             // expand not 'safe' characters into %XX, spaces to +s
             byte[] expandedBytes = new byte[count + cUnsafe * 2];
-            int pos = 0;
+            GetEncodedBytes(bytes, offset, count,expandedBytes);
+            return expandedBytes;
+        }
 
-            for (int i = 0; i < count; i++)
+        private static void GetEncodedBytes(byte[] originalBytes, int startIndex, int count, byte[] newBytes)
```

Comment:
Nit: can we use the name offset instead of startIndex?  That'll help to be consistent with the call sites and typical usage in APIs like this.